### PR TITLE
Support for autodetection Calico node IP from k8s InternalIP

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Adds config support for the featuer added in this PR: https://github.com/projectcalico/node/pull/1242

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: api/v1/installation_types.go [hunk 1]
```diff
@@ -368,6 +368,11 @@ type NodeAddressAutodetection struct {
 	// +optional
 	FirstFound *bool `json:"firstFound,omitempty"`
 
+	// Kubernetes configures Calico to detect node addresses based on the Kubernetes API.
```
Comment (caseydavenport) [inline api/v1/installation_types.go:371]: @tmjd @Brian-McM what do you think about this API? Please give it a critical eye!

The reason behind not having it as a bool is that we might want to extend this to other options in the future. For example. any of these: https://kubernetes.io/docs/concepts/architecture/nodes/#addresses

Or our own modification - e.g., `NodeCalicoIP` - which looks for an address in the list with type `CalicoIP`

File: api/v1/installation_types.go [hunk 2]
```diff
@@ -368,6 +368,11 @@ type NodeAddressAutodetection struct {
...
+	// +optional
+	// +kubebuilder:validation:Enum=InternalIP
```
Comment (tmjd) [inline api/v1/installation_types.go:373]: This validation does not match `NodeInternalIP`

File: api/v1/installation_types.go (hunk 1 above)
Comment (tmjd) [reply api/v1/installation_types.go:371]: I'm always a fan of not using a bool in the API, I think that's a good choice.
`kubernetes` seems very broad/generic but I think the possible values (and the comments on them) should clarify any ambiguity.

File: pkg/controller/migration/convert/network.go [hunk 3]
```diff
@@ -425,13 +426,21 @@ func handleAutoDetectionMethod(c *components, install *operatorv1.Installation)
...
+	// cidr=
 	if strings.HasPrefix(*method, AutodetectionMethodCIDR) {
 		ifStr := strings.TrimPrefix(*method, AutodetectionMethodCIDR)
 		cidrs := strings.Split(ifStr, ",")
 		install.Spec.CalicoNetwork.NodeAddressAutodetectionV4 = &operatorv1.NodeAddressAutodetection{CIDRS: cidrs}
 		return nil
 	}
 
+	// kubernetes-internal-ip
+	if *method == "" || *method == AutodetectionMethodNodeIP {
```
Comment (tmjd) [inline pkg/controller/migration/convert/network.go:438]: Nice job adding migration code. WDYT about adding a test for it?

**Review by tmjd — COMMENTED** [review]
Looks good, one change needed, maybe another test but it mainly looks good.

**Pull request merged**